tools/evm/Program.cs

```diff
@@ -0,0 +1,98 @@
+using System.CommandLine;
+using Evm.JsonTypes;
+using Evm.T8NTool;
+using Nethermind.Core.Exceptions;
+using Nethermind.Core.Extensions;
+using Nethermind.Evm;
+using Nethermind.Serialization.Rlp;
+using Nethermind.Specs.Forks;
+
+namespace Evm
+{
+    public static class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            var rootCmd = new RootCommand();
+            rootCmd.Name = "evm";
+            ConfigureT8NCommand(ref rootCmd);
+
+            await rootCmd.InvokeAsync(args);
+        }
+
+        static void ConfigureT8NCommand(ref RootCommand rootCmd)
+        {
+            var inputAllocOpt = new Option<string>("--input.alloc", description: "Input allocations", getDefaultValue: () => "alloc.json");
+            var inputEnvOpt = new Option<string>("--input.env", description: "Input environment", getDefaultValue: () => "env.json");
+            var inputTxsOpt = new Option<string>("--input.txs", description: "Input transactions", getDefaultValue: () => "txs.json");
+            var outputAllocOpt = new Option<string>("--output.alloc", description: "Output allocations");
+            var outputBaseDirOpt = new Option<string>("--output.basedir", description: "Output base directory");
+            var outputBodyOpt = new Option<string>("--output.body", description: "Output body");
+            var outputResultOpt = new Option<string>("--output.result", description: "Output result");
+            var stateChainIdOpt = new Option<ulong>("--state.chainId", description: "State chain id", getDefaultValue: () => 1);
+            var stateForkOpt = new Option<string>("--state.fork", description: "State fork", getDefaultValue: () => "GrayGlacier");
+            var stateRewardOpt = new Option<string>("--state.reward", description: "State reward");
+            var traceMemoryOpt = new Option<bool>("--trace.memory", description: "Trace memory", getDefaultValue: () => false);
+            var traceOpt = new Option<bool>("--trace", description: "Configures the use of the JSON opcode tracer. This tracer emits traces to files as trace-<txIndex>-<txHash>.jsonl", getDefaultValue: () => false);
+            var traceNoStackOpt = new Option<bool>("--trace.noStack", description: "Trace no stack", getDefaultValue: () => false);
+            var traceReturnDataOpt = new Option<bool>("--trace.returnData", description: "Trace return data", getDefaultValue: () => false);
```

Comment:
Would it possible to use strict lower-case for the cli flags? Or make them case-insensitive?

We can accommodate variations, but until now all other tools supported by EEST use lowercase as apposed to camel case (see also `basedir`), which makes it easier to abstract the interface for all tools.

```suggestion
            var inputAllocOpt = new Option<string>("--input.alloc", description: "Input allocations", getDefaultValue: () => "alloc.json");
            var inputEnvOpt = new Option<string>("--input.env", description: "Input environment", getDefaultValue: () => "env.json");
            var inputTxsOpt = new Option<string>("--input.txs", description: "Input transactions", getDefaultValue: () => "txs.json");
            var outputAllocOpt = new Option<string>("--output.alloc", description: "Output allocations");
            var outputBaseDirOpt = new Option<string>("--output.basedir", description: "Output base directory");
            var outputBodyOpt = new Option<string>("--output.body", description: "Output body");
            var outputResultOpt = new Option<string>("--output.result", description: "Output result");
            var stateChainIdOpt = new Option<ulong>("--state.chainid", description: "State chain id", getDefaultValue: () => 1);
            var stateForkOpt = new Option<string>("--state.fork", description: "State fork", getDefaultValue: () => "GrayGlacier");
            var stateRewardOpt = new Option<string>("--state.reward", description: "State reward");
            var traceMemoryOpt = new Option<bool>("--trace.memory", description: "Trace memory", getDefaultValue: () => false);
            var traceOpt = new Option<bool>("--trace", description: "Configures the use of the JSON opcode tracer. This tracer emits traces to files as trace-<txIndex>-<txHash>.jsonl", getDefaultValue: () => false);
            var traceNoStackOpt = new Option<bool>("--trace.nostack", description: "Trace no stack", getDefaultValue: () => false);
            var traceReturnDataOpt = new Option<bool>("--trace.returndata", description: "Trace return data", getDefaultValue: () => false);
```